website/content/docs/upgrading/plugins.mdx

```diff
@@ -6,206 +6,184 @@ description: These are general upgrade instructions for Vault plugins.
-register a new plugin atop the built-in one and leave any mounts alone.
-
-### Procedure for Overriding Built-in Plugins
-
-The syntax is the same as an external plugin, with the difference being you
-name it the same as a built-in:
-
-```shell-session
-$ vault plugin register \
-  -sha256=<SHA256 Hex value of the plugin binary> \
-  -command=azure_new \
-  secret \
-  azure
-```
+1. [Register][plugin_registration] the first version of your plugin to the catalog.
+   Skip this step if your initial plugin is built-in or already registered.
 
-"-command=azure_new" is the name of the binary, "secret" is the plugin type,
-and "azure" is the name of the built-in plugin that we're overriding. We can
-verify that the override is in place:
+    ```shell-session
+    $ vault plugin register
+        -sha256=<SHA256 Hex value of the plugin binary> \
+        database \
+        my-db-plugin
+    ```
 
-```shell-session
-$ vault plugin info secret azure
-Key        Value
----        -----
-args       []
-builtin    false
-command    azure_new
-name       azure
-sha256     f6f6ec45d37484c257aa9ff80444b9f244aaef1c650edf8a42a2a1d3f00db2c5
-```
+2. [Mount][plugin_management] the plugin. Skip this step if your initial plugin
+   is already mounted.
 
-At this point we've overridden the built-in, but it is not yet actively
-handling requests. For that we run:
+    ```shell-session
+    $ vault secrets enable database
+    $ vault write database/config/my-db \
+        plugin_name=my-db-plugin \
+        # ...
+    ```
 
-```shell-session
-$ vault plugin reload -plugin=azure
-```
+3. Register a second version of your plugin. You **must** use the same plugin
+   type and name (the last two arguments) as the plugin being upgraded. This is
+   true regardless of whether the plugin being upgraded is built-in or external.
 
-### Procedure for Reverting After Overriding A Built-in Plugin
+    ```shell-session
+    $ vault plugin register \
+        -sha256=<SHA256 Hex value of the plugin binary> \
+        -command=my-db-plugin-1.0.1 \
+        -version=v1.0.1 \
+        database \
+        my-db-plugin
+    ```
 
-To revert the override, first deregister the plugin:
+4. Update the database config with the new version. The database secrets
+   engine will immediately reload the plugin, using the new version. Any omitted
+   config parameters will not be updated.
+    
+    ```shell-session
+    $ vault write database/config/my-db \
+        plugin_version=v1.0.1
+    ```
 
-```shell-session
-$ vault plugin deregister secret azure
-```
+Until step 4, the mount will still run the first version of `my-db-plugin`. When
+the reload is triggered, Vault will kill `my-db-plugin`’s process and start the
+new plugin process for `my-db-plugin` version 1.0.1.
 
-Next, verify the override has been reverted and we are now using the built-in
-plugin:
+-> **Important:** Plugin reload of a new plugin binary must be
```

Comment:
Note to self: I still need to either update this note, or add an extra step to reload globally.

Reply:
I tested this out, and when the config is updated, performance replicas automatically replicate the db config and so automatically reload the plugins too as expected - no extra steps required.